src/Microsoft.DotNet.VersionTools/Automation/PullRequestCreator.cs

```diff
@@ -0,0 +1,201 @@
+                {
+                    pullRequestToUpdate = await client.SearchPullRequestsAsync(
+                        Upstream,
+                        BranchPrefix,
+                        _auth.User);
+
+                    if (pullRequestToUpdate == null)
+                    {
+                        Trace.TraceInformation($"No existing pull request found.");
+                    }
+                    else
+                    {
+                        Trace.TraceInformation(
+                            $"Pull request already exists for {BranchPrefix} in {Upstream.Segments}. " +
+                            $"#{pullRequestToUpdate.Number}, '{pullRequestToUpdate.Title}'");
+
+                        string blockedReason = GetUpdateBlockedReason(client, pullRequestToUpdate);
+
+                        if (blockedReason == null)
+                        {
+                            originBranch = new GitHubBranch(
+                                pullRequestToUpdate.Head.Ref,
+                                originBranch.Project);
+                        }
+                        else
+                        {
+                            string comment =
+                                $"Couldn't update this pull request: {blockedReason}\n" +
+                                $"Would have applied '{commitMessage}'";
+
+                            await client.PostCommentAsync(Upstream, pullRequestToUpdate.Number, comment);
+                            return;
+                        }
+                    }
+                }
+
+                bool forcePush = pullRequestToUpdate != null;
+
+                PushNewCommit(
+                    originBranch,
+                    commitMessage,
+                    forcePush);
+
+                if (forcePush)
+                {
+                    await client.UpdateGitHubPullRequestAsync(Upstream, pullRequestToUpdate.Number, title, description);
+                }
+                else
+                {
+                    await client.PostGitHubPullRequestAsync(title, description, originBranch, UpstreamBranch);
+                }
+            }
+        }
+
+        public static string NotificationString(IEnumerable<string> usernames)
+        {
+            return $"/cc @{string.Join(" @", usernames)}";
+        }
+
+        private string GetUpdateBlockedReason(GitHubClient client, GitHubPullRequest pullRequest)
+        {
+            if (pullRequest.Head.User.Login != Origin.Owner)
+            {
+                return $"Owner of head repo '{pullRequest.Head.User.Login}' is not '{Origin.Owner}'";
+            }
+            if (!pullRequest.Head.Ref.StartsWith(BranchPrefix))
+            {
+                return $"Ref name '{pullRequest.Head.Ref}' does not start with '{BranchPrefix}'";
+            }
+
+            GitCommit commit = client.GetCommitAsync(Origin, pullRequest.Head.Sha).Result;
+            if (commit.Author.Name != GitAuthorName)
```

Comment:
This is here to help with exactly that scenario, so that the bot doesn't overwrite (with `push force`) the fixup changes someone's making. (This doesn't block the PR from being merged or anything.)